flexmeasures/data/services/sensors.py

```diff
@@ -97,82 +122,102 @@ def get_staleness(
-        else:
-            # Default to status specs indicating immediate staleness after knowledge time
-            status_specs = {"staleness_search": {}, "max_staleness": "PT0H"}
+    status_specs = sensor.attributes.get("status_specs", dict())
+    if status_specs:
+        return status_specs
+
+    status_specs["staleness_search"] = {}
+    # Consider forecast or schedule data stale if it is less than 12 hours in the future
+    status_specs["max_future_staleness"] = "-PT12H"
+
+    # Default to status specs for economical sensors with daily updates
+    if sensor.knowledge_horizon_fnc == "x_days_ago_at_y_oclock":
+        status_specs["max_staleness"] = "P1D"
+    else:
+        # Default to status specs indicating immediate staleness after knowledge time
+        status_specs["max_staleness"] = "PT0H"
     return status_specs
 
 
-def get_status(
+def get_statuses(
     sensor: Sensor,
     now: datetime,
     status_specs: dict | None = None,
-) -> dict:
-    """Get the status of the sensor
+) -> list[dict]:
+    """Get the status of the sensor by source type.
     Main part of result here is a stale value, which is True if the sensor is stale, False otherwise.
     Other values are just context information for the stale value.
     """
     if status_specs is None:
         status_specs = get_status_specs(sensor=sensor)
     status_specs = StatusSchema().load(status_specs)
     max_staleness = status_specs.pop("max_staleness")
+    max_future_staleness = status_specs.pop("max_future_staleness")
     staleness_search = status_specs.pop("staleness_search")
-    staleness: timedelta = get_staleness(
+    stalenesses = get_stalenesses(
         sensor=sensor,
         staleness_search=staleness_search,
         now=now,
     )
-    if staleness is not None:
-        staleness_since = now - staleness
-        stale = staleness > max_staleness
-        reason = (
-            "" if stale else "not "
-        ) + f"more than {naturaldelta(max_staleness)} old"
-        staleness = staleness if staleness > timedelta(0) else -staleness
-    else:
-        staleness_since = None
-        stale = True
-        reason = "no data recorded"
-    status = dict(
-        staleness=staleness,
-        stale=stale,
-        staleness_since=staleness_since,
-        reason=reason,
-    )
-    return status
+
+    statuses = list()
+    for source, staleness in (stalenesses or {None: None}).items():
+        if staleness is None:
+            staleness_since = None
+            stale = True
+            reason = "no data recorded"
+        else:
+            max_source_staleness = (
+                max_staleness if staleness > timedelta(0) else max_future_staleness
+            )
+            staleness_since = now - staleness
+            stale = staleness > max_source_staleness
+            comparison = "more" if staleness > timedelta(0) else "less"
+            timeline = "old" if staleness > timedelta(0) else "in the future"
+            reason = (
```

Comment:
Maybe the wording could be more helpful if we include the staleness.

For example: "most recent data is from 2 hours ago, but should not be more than a moment old"